server/src/main/java/org/apache/druid/metadata/IndexerSQLMetadataStorageCoordinator.java

```diff
@@ -1605,7 +1599,13 @@ protected DataStoreMetadataUpdateResult updateDataSourceMetadataWithHandle(
                                 .bind("commit_metadata_sha1", newCommitMetadataSha1)
                                 .execute();
 
-      retVal = numRows == 1 ? DataStoreMetadataUpdateResult.SUCCESS : DataStoreMetadataUpdateResult.TRY_AGAIN;
+      retVal = numRows == 1
+          ? DataStoreMetadataUpdateResult.SUCCESS
+          : new DataStoreMetadataUpdateResult(
+              true,
+          true,
+          "Failed to insert metadata for datasource [%s]",
```

Comment:
This error is not very actionable for newbies: it is helpful only to developers already familiar with the area. If we can't insert metadata, then we've probably got a fatal or data loss error, don't we?

Perhaps:

> Fatal error: Supervisor %s failed to update metadata for datasource %s in table %s. Newly ingested segments are lost.

Or some such.

Reply:
Nit: the `word[%s]` pattern is a bit awkward. It may have started from thread names: `datasource[%s]`. But, for error messages, a better approach is:

* If the `%s` value is a single symbol (no spaces), omit the brackets.
* If the `%s` value contains spaces, include brackets, but insert a space before the bracket.

Example: `datasource %s not found`, `The name [%s] is not valid`

In the first case, the rules for datasource names disallow spaces. In the second, the name is invalid, so it may well be "foo bar" (with spaces) so we need the brackets.